lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -44,21 +44,36 @@ def object
         def initialize(serializer, options = {})
           super
           @include_tree = IncludeTree.from_include_args(options[:include])
-          @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options.delete(:fields))
+          @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options[:fields])
         end
 
+        # Build JSON API document.
         def serializable_hash(options = nil)
           options ||= {}
 
-          hash =
-            if serializer.respond_to?(:each)
-              serializable_hash_for_collection(options)
+          resources = resource_objects_for(serializer)
+          primary_data, included = resources.each_with_object([[], []]) do |(_, resource), (prim, incl)|
```

Comment:
so this is building two arrays (hence the `[[], []]`), from `resources` which is a Hash of resources keyed by something we don't care about (hence the `_`).